client/src/stores/userStore.ts

```diff
@@ -59,8 +59,14 @@ export const useUserStore = defineStore("userStore", () => {
             loadPromise = getCurrentUser()
                 .then(async (user) => {
                     const historyStore = useHistoryStore();
-                    currentUser.value = { ...user, isAnonymous: !user.email };
-                    currentPreferences.value = user?.preferences ?? null;
+                    if ('email' in user) {
+                        currentUser.value = { ...user, isAnonymous: false };
+                        currentPreferences.value = user.preferences;
+                    }
+                    else {
+                        currentUser.value = { isAnonymous: true };
+                        currentPreferences.value = null;
+                    }
```

Comment:
I would also recommend dropping the `User` and `AnonymousUser` interfaces declared at the top and directly using the ones provided now by the nice generated client schema like:

```ts
type AnonymousUser = components["schemas"]["AnonUserModel"];
```